### PR TITLE
Migration of recordingeditor and recordingfields to ES6 modules

### PR DESCRIPTION
**Pull request by Camc314**
**Issues**
part of #868

### CODE REVIEW

File: src/components/recordingcreator/recordingeditor.js [hunk 1]
```diff
@@ -1,150 +1,150 @@
...
+
+function init(context) {
+    context.querySelector('.btnCancel').addEventListener('click', function () {
+        closeDialog(false);
+    });
+
+    context.querySelector('.btnCancelRecording').addEventListener('click', function () {
+        const apiClient = connectionManager.getApiClient(currentServerId);
+
+        (apiClient, currentItemId).then(function () {
```
Comment (dmitrylyzo) [inline src/components/recordingcreator/recordingeditor.js:61]: What's going on here?

Comment (Camc314) [reply src/components/recordingcreator/recordingeditor.js:61]: Looks like I deleted a function, probably when rebasing :/

Comment (Camc314) [reply src/components/recordingcreator/recordingeditor.js]: Should be fixed now.

File: src/components/recordingcreator/recordingcreator.js [hunk 2]
```diff
@@ -184,7 +184,13 @@ function showEditor(itemId, serverId) {
...
             init(dlg);
 
-            reload(dlg, itemId, serverId);
+            currentRecordingFields = new recordingFields({
+                parent: dlg.querySelector('.recordingFields'),
+                programId: itemId,
+                serverId: serverId
+            });
+
+            events.on(currentRecordingFields, 'recordingchanged', onRecordingChanged);
```
Comment (dmitrylyzo) [inline src/components/recordingcreator/recordingcreator.js:193]: bad merge?

Comment (Camc314) [reply src/components/recordingcreator/recordingcreator.js]: Yup.

**Pull request merged**